core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/batchutilities/BatchBufferTest.java

```diff
@@ -0,0 +1,269 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.batchutilities;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.testng.Assert;
+
+public class BatchBufferTest {
+
+    private BatchManager<String, String, List<RequestWithId>> buffer;
+    private String destination;
+
+    BatchAndSendFunction<String, List<RequestWithId>> batchingFunction =
+        (identifiedRequests, destination) -> {
+            List<RequestWithId> entries = new ArrayList<>(identifiedRequests.size());
+            identifiedRequests.forEach(identifiedRequest -> {
+                String id = identifiedRequest.getId();
+                String request = identifiedRequest.getRequest();
+                entries.add(new RequestWithId(id, request));
+            });
+            return CompletableFuture.supplyAsync(() -> {
+                waitForTime(150);
+                return entries;
+            });
+        };
+
+    UnpackBatchResponseFunction<List<RequestWithId>, String> unpackResponseFunction =
+        requestBatchResponse -> {
+            List<IdentifiedResponse<String>> identifiedResponses = new ArrayList<>();
+            for (RequestWithId requestWithId : requestBatchResponse) {
+                identifiedResponses.add(new IdentifiedResponse<>(requestWithId.getId(), requestWithId.getMessage()));
+            }
+            return identifiedResponses;
+        };
+
+    //Object to mimic a batch request entry
+    private static class RequestWithId {
+
+        private final String id;
+        private final String message;
+
+        public RequestWithId(String id, String message) {
+            this.id = id;
+            this.message = message;
+        }
+
+        public String getId() {
+            return id;
+        }
+
+        public String getMessage() {
+            return message;
+        }
+    }
+
+    @Before
+    public void beforeEachBufferTest() {
+        buffer = new BatchManager<>(10, Duration.ofMillis(200), batchingFunction, unpackResponseFunction);
+        destination = "testDestination";
+    }
+
+    @After
+    public void afterEachBufferTest() {
+        buffer.close();
+    }
+
+    @Test
+    public void sendTenRequestsTest() {
```

Comment:
nit, `Test` seems redundant. We don't have a consistent naming convention for unit tests in v2, but most of them follow similar format of methodUnderTest_when_expectedBehavior I think